warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/TableSplitsCache.java

```diff
@@ -35,25 +40,68 @@ public class TableSplitsCache extends BaseHdfsFileCacheUtil {
     
     public static final String REFRESH_SPLITS = "datawave.ingest.refresh.splits";
     public static final String SPLITS_CACHE_DIR = "datawave.ingest.splits.cache.dir";
+    public static final String SPLITS_CACHE_FILE = "datawave.ingest.splits.cache.fileName";
+    
     public static final String MAX_SPLIT_DECREASE = "datawave.ingest.splits.max.decrease.number";
     public static final String MAX_SPLIT_PERCENTAGE_DECREASE = "datawave.ingest.splits.max.decrease.percentage";
     private static final Logger log = Logger.getLogger(TableSplitsCache.class);
     private static final String DEFAULT_SPLITS_CACHE_DIR = "/data/splitsCache";
-    private static final String SPLITS_CACHE_FILE = "all-splits.txt";
+    public static final String DEFAULT_SPLITS_CACHE_FILE = "all-splits.txt";
     private static final short DEFAULT_MAX_SPLIT_DECREASE = 42;
     private static final double DEFAULT_MAX_SPLIT_PERCENTAGE_DECREASE = .5;
     private static final boolean DEFAULT_REFRESH_SPLITS = true;
+    private static final String NO_LOCATION = "noloc";
+    private static TableSplitsCache cache;
+    
     private Path splitsPath = null;
-    private Map<String,List<Text>> splits = null;
+    private Map<String,List<Text>> splits = new HashMap<>();
+    private Map<String,Map<Text,String>> splitLocations = new HashMap<>();
+    
+    private Map<Text,String> getSplitsWithLocation(String table) throws AccumuloException, AccumuloSecurityException, TableNotFoundException {
+        SortedMap<KeyExtent,String> tabletLocations = new TreeMap<>();
+        SortedMap<Text,String> tabletLocationsByEndRow = new TreeMap<>();
+        
```

Comment:
```
        tabletLocationsByEndRow = tabletLocations.entrySet().stream().
                collect(Collectors.toMap(e -> e.getKey().getEndRow(),
                        e -> e.getValue() == null ? NO_LOCATION : e.getValue(),
                        (o1, o2) -> o1,
                        TreeMap::new));
```

Reply:
throws an NPE

Reply:
Added a filter for endRow not null.  1ms slower on average than the regular for loop.  Not a big deal, but not sure I see the appeal.